src/goto-programs/goto_convert.cpp

```diff
@@ -344,14 +344,49 @@ void goto_convertt::convert_label(
 
   goto_programt tmp;
 
-  // magic thread creation label?
+  // magic thread creation label, see ansi-c/library/pthread_lib.c
   if(has_prefix(id2string(label), "__CPROVER_ASYNC_"))
   {
-    // this is like a START_THREAD
-    codet tmp_code(ID_start_thread);
-    tmp_code.copy_to_operands(code.op0());
-    tmp_code.add_source_location()=code.source_location();
-    convert(tmp_code, tmp);
+    // A: START_THREAD : C
```

Comment:
:+1: for documenting the GOTO being generated, if you could label it as such and explain what the block does

```C++
// Generate GOTO to ...
// A: START_THREAD : C
// ...
```

Reply:
Or consider pulling this generation out into a function that returns a `code_blockt` which allows you to document it more clearly.